forms-flow-web/src/components/ServiceFlow/list/sort/CreateNewFilter.js

```diff
@@ -0,0 +1,470 @@
+      <Divider />
+      <List>
+        <h5 style={{ fontWeight: "bold", fontSize: "18px" }}>
+          Criteria <i className="fa fa-info-circle"></i>{" "}
+        </h5>
+        <h5 style={{ fontSize: "18px" }}>Definition Key</h5>
+        <span
+          style={{
+            textDecoration: "underline",
+            fontSize: "14px",
+            cursor: "pointer",
+          }}
+          onClick={() => handleSpanClick(1)}
+          className="px-1 py-1"
+        >
+          <i className="fa fa-plus-circle" style={{ marginRight: "6px" }} />
+          Add Value
+        </span>
+        {inputVisibility[1] && (
+          <input
+            type="text"
+            className="Criteria-addValueInputBox"
+            value={definitionKeyId}
+            onChange={(e) => setDefinitionKeyId(e.target.value)}
+          />
+        )}
+        <h5>Candidate Group</h5>
+        <span
+          style={{
+            textDecoration: "underline",
+            fontSize: "14px",
+            cursor: "pointer",
+          }}
+          onClick={() => handleSpanClick(2)}
+          className="px-1 py-1"
+        >
+          <i className="fa fa-plus-circle" style={{ marginRight: "6px" }} />
+          Add Value
+        </span>
+        {inputVisibility[2] && (
+          <input
+            type="text"
+            className="Criteria-addValueInputBox"
+            value={candidateGroup}
+            onChange={(e) => setCandidateGroup(e.target.value)}
+          />
+        )}
+        <h5>Asignee</h5>
+        <span
+          style={{
+            textDecoration: "underline",
+            fontSize: "14px",
+            cursor: "pointer",
+          }}
+          onClick={() => handleSpanClick(3)}
+          className="px-1 py-1"
+        >
+          <i className="fa fa-plus-circle" style={{ marginRight: "6px" }} />
+          Add Value
+        </span>
+        {inputVisibility[3] && (
+          <input
+            type="text"
+            className="Criteria-addValueInputBox"
+            value={taskName}
+            onChange={(e) => setTaskName(e.target.value)}
+          />
+        )}
+
+        <div
+          style={{ display: "flex", alignItems: "center", marginTop: "10px" }}
+        >
+          <input
+            type="checkbox"
+            id="assignedTask-checkbox"
+            checked={includeAssignedTasks}
+            onChange={includeAssignedTasksCheckboxChange}
+            style={{ marginRight: "6px" }}
+          />
+          <h5 style={{ fontSize: "18px", marginBottom: "3px" }}>
+            Include Assigned Task
+          </h5>
+        </div>
+        <Divider />
+        <List>
+          <h5 style={{ fontWeight: "bold", fontSize: "18px" }}>
+            Variable <i className="fa fa-info-circle"></i>
+          </h5>
+
+          <div style={{ display: "flex", alignItems: "center" }}>
+            <input
+              type="checkbox"
+              id="my-checkbox"
+              checked={showUndefinedVariable}
```

Comment:
what is undefinedVariable means

Reply:
label name is "show Undefined Variable" so the variable also given as "showUndefinedVariable".